yarn-project/archiver/src/archiver/memory_archiver_store/memory_archiver_store.ts

```diff
@@ -206,17 +211,35 @@ export class MemoryArchiverStore implements ArchiverDataStore {
   addLogs(blocks: L2Block[]): Promise<boolean> {
     blocks.forEach(block => {
       this.noteEncryptedLogsPerBlock.set(block.number, block.body.noteEncryptedLogs);
+      block.body.noteEncryptedLogs.txLogs.forEach(txLogs => {
+        const noteLogs = txLogs.unrollLogs();
+        noteLogs.forEach(noteLog => {
+          const tag = new Fr(noteLog.data.subarray(0, 32));
```

Comment:
Also is it not possible for `new Fr` to fail? Fields are less than 32 bytes, so I'd imagine `new Fr('0xffff...ffff)` doesn't work.

Reply:
Protected all `addLogs`, but I was under the impression that we couldn't have logs that small. 

`new Fr(...)` will fail if you provide a number larger than the modulus, so that's covered.

Reply:
Sorry, my original message wasn't clear. I meant that `new Fr` would likely throw, and we don't have any kind of error handling. This means that if a single log in a block contains a log with the first 32 bytes having a value larger than the modulus, then the entire block would be unprocessed by the node.

What we're doing here is _attempting_ to index logs by their tag, but not all logs might have a tag. So we need a way to handle these cases without preventing further processing of remaining logs.

Reply:
Understood, protected and tested!